sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/RedundantCast.cs

```diff
@@ -106,19 +106,24 @@ private static void CheckExtensionMethodInvocation(SyntaxNodeAnalysisContext con
                 return;
             }
 
-            var elementType = GetElementType(invocation, methodSymbol, context.SemanticModel);
-            if (elementType == null)
+            if (!(methodSymbol.ReturnType is INamedTypeSymbol returnType) ||
+                !returnType.TypeArguments.Any())
             {
                 return;
             }
 
-            if (!(methodSymbol.ReturnType is INamedTypeSymbol returnType) ||
-                !returnType.TypeArguments.Any())
+            var castType = returnType.TypeArguments.First();
+            if (methodSymbol.Name == "OfType"
+                && (castType.IsReferenceType || castType.Name == "Nullable"))
             {
                 return;
             }
 
-            var castType = returnType.TypeArguments.First();
+            var elementType = GetElementType(invocation, methodSymbol, context.SemanticModel);
```

Comment:
as a side effect, less semantic model queries will be done

Reply:
Indeed, it is better to call this here but on the other hand we do get symbols for all the method invocations when we could filter them first by name.